core-build/gulp-core-build-typescript/src/miniTasks/BaseCmdTask.ts

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as childProcess from 'child_process';
+
+import { GulpTask } from '@microsoft/gulp-core-build';
+
+export interface ICallCmdOptions {
+  onData: (chunk: Buffer) => void;
+  onError: (chunk: Buffer) => void;
+  onClose: (code: number, hasErrors: boolean, resolve: () => void, reject: (error: Error) => void) => void;
+}
+
+/**
+ * @alpha
+ */
+export abstract class BaseCmdTask<TTaskConfig> extends GulpTask<TTaskConfig> {
+  protected _callCmd(
+    binaryPath: string,
+    cwd: string,
+    args: string[],
+    options?: Partial<ICallCmdOptions>
+  ): Promise<void> {
+    options = options || {};
+    const promiseHandlers: ICallCmdOptions = {} as ICallCmdOptions;
+
+    promiseHandlers.onData = options.onData
+      ? options.onData
+      : (data: Buffer) => {
+          this._logBuffer(data, this.log);
```

Comment:
>this._logBuffer(data, this.log); [](start = 10, length = 32)

this.log(data.toString().trim());

Reply:
I don't buy that _logBuffer is really necessary

---
In reply to: [199006445](https://github.com/Microsoft/web-build-tools/pull/729#discussion_r199006445) [](ancestors = 199006445)